src/devices/src/virtio/block/device.rs

```diff
@@ -272,7 +272,11 @@ impl Block {
                     len = 0;
                 }
             }
-            queue.add_used(mem, head.index, len);
+
+            queue.add_used(mem, head.index, len).unwrap_or_else(|e| {
+                error!("Failed to add available descriptor {}: {}", head.index, e);
+                std::process::exit(1);
```

Comment:
I wouldn't exit on an `add_used()` failure. It can be a transitory error, a corrupted descriptor, etc. The process might continue to work correctly.

Reply:
I was considering the possibility for this to fail due to a malicious guest/attacker, in which case the process should not continue to work past this point. WDYT?

Reply:
Like Serban said, we shouldn't `exit()` on these events. If the error cannot be cascaded up (like in these device functions) we should log the error and move on. The invalid access is blocked and logged.

Reply:
Handled it.